infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationEventListener.java

```diff
@@ -0,0 +1,58 @@
+package org.corfudb.infrastructure.logreplication.infrastructure;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.logreplication.proto.LogReplicationMetadata.ReplicationEvent;
+
+import org.corfudb.runtime.collections.CorfuStreamEntries;
+import org.corfudb.runtime.collections.CorfuStreamEntry;
+import org.corfudb.runtime.collections.StreamListener;
+
+import java.util.List;
+
+@Slf4j
+public class LogReplicationEventListener implements StreamListener {
+    private CorfuReplicationDiscoveryService discoveryService;
+
+    public  LogReplicationEventListener(CorfuReplicationDiscoveryService discoveryService) {
+        this.discoveryService = discoveryService;
+    }
+
+    public void start() {
+        discoveryService.getLogReplicationMetadataManager().subscribeReplicationEventTable(this);
+    }
+
+    public void stop() {
+        discoveryService.getLogReplicationMetadataManager().unsubscribeReplicationEventTable(this);
+    }
+
+    @Override
+    public void onNext(CorfuStreamEntries results) {
+        /**
+         * If the current node is not a leader, ignore the notifications.
+         */
+        if (!discoveryService.getIsLeader().get()) {
+            log.info("The onNext call with {} will be skipped as the current node {}  in the cluster {} is not the leader.",
+                    results, discoveryService.getLocalNodeDescriptor(), discoveryService.getLocalClusterDescriptor());
+            return;
+        }
+
+        log.info("LogReplicationEventListener onNext {} will be processed at node {} in the cluster {}",
```

Comment:
this message will be printed as a huge blob and anyways it will only be printed on the corfu.logs of the node receiving it, we can likely remove the localNodeDescriptor and localClusterDescriptor